instrumentation/opentelemetry-instrumentation-falcon/src/opentelemetry/instrumentation/falcon/__init__.py

```diff
@@ -138,14 +145,16 @@ def instrumentation_dependencies(self) -> Collection[str]:
         return _instruments
 
     def _instrument(self, **kwargs):
-        self._original_falcon_api = falcon.API
-        falcon.API = partial(_InstrumentedFalconAPI, **kwargs)
+        self._original_falcon_api = getattr(falcon, _instrument_app)
+        setattr(
+            falcon, _instrument_app, partial(_InstrumentedFalconAPI, **kwargs)
+        )
 
     def _uninstrument(self, **kwargs):
-        falcon.API = self._original_falcon_api
+        setattr(falcon, _instrument_app, self._original_falcon_api)
 
 
-class _InstrumentedFalconAPI(falcon.API):
+class _InstrumentedFalconAPI(getattr(falcon, _instrument_app)):
```

Comment:
I really dislike using a `getattr` in a class def, but it works and it avoids a deprecation warning for Falcon 3 users.